# 📌 课程作业：HW1（总分 100 分）

## **提交说明**
📩 **提交方式**：请将作业发送至邮箱 **jiqing@cup.edu.cn**

📌 **邮件主题格式**：
```ECO-HW1-姓名-学号```

📂 **文件命名格式**：
```ECO_HW1_姓名_学号.ipynb```

📅 **截止提交日期**：
👉 **2025 年 3 月 24 日（星期一上课前）**

---

## **求解模型**
我们研究的模型如下：
$
\ln(wage) = \beta_0 + \beta_1 \cdot educ + \beta_2 \cdot exper + \beta_3 \cdot expersq + \epsilon
$

- 其中，$educ$ 为 **内生变量**，其 **工具变量** 选择为：
  - $constant$（常数项）
  - $exper$（工作经验）
  - $expersq$（经验的平方）
  - $motheduc$（母亲受教育程度）
  - $fatheduc$（父亲受教育程度）



## **数据导入与预处理**
本次作业使用 **Wooldridge 经济学数据集**，请确保已安装 `wooldridge` 库，并使用以下代码导入 **Mroz** 数据集：

---

In [1]:
import wooldridge as woo
mroz = woo.dataWoo('mroz') # 加载 Mroz 数据集
mroz = mroz.dropna(subset=['lwage']) # 删除缺失工资数据的观测值
mroz.head() # 显示前几行数据

inlf  hours  kidslt6  kidsge6  age  educ    wage  repwage  hushrs  husage  \
0     1   1610        1        0   32    12  3.3540     2.65    2708      34   
1     1   1656        0        2   30    12  1.3889     2.65    2310      30   
2     1   1980        1        3   35    12  4.5455     4.04    3072      40   
3     1    456        0        3   34    12  1.0965     3.25    1920      53   
4     1   1568        1        2   31    14  4.5918     3.60    2000      32   

   ...   faminc     mtr  motheduc  fatheduc  unem  city  exper   nwifeinc  \
0  ...  16310.0  0.7215        12         7   5.0     0     14  10.910060   
1  ...  21800.0  0.6615         7         7  11.0     1      5  19.499981   
2  ...  21040.0  0.6915        12         7   5.0     0     15  12.039910   
3  ...   7300.0  0.7815         7         7   5.0     0      6   6.799996   
4  ...  27300.0  0.6215        12        14   9.5     1      7  20.100058   

      lwage  expersq  
0  1.210154      196  
1  0.328512       25  
2  1.514138      225  
3  0.092123       36  
4  1.524272       49  

[5 rows x 22 columns]

### **任务一：计算两阶段最小二乘的系数方差和 T 值（共 50 分）**

---

### **两阶段最小二乘法（2SLS）**
**直接估计公式**：
$
\beta = (X'Z(Z'Z)^{-1}Z'X)^{-1} X'Z(Z'Z)^{-1}Z'y
$

**估计结果**：
$
\hat{\ln(wage)} = 0.0481 + 0.0614 \cdot educ + 0.0442 \cdot exper - 0.0009 \cdot expersq
$

---

#### **任务 1.1：计算系数估计值**
- 估计 **$\beta$ 系数**，即使用 2SLS 方法求解 **回归系数估计值**（10 分）

#### **任务 1.2：假设误差项服从独立同方差**
- 计算 **系数的方差** 和 **T 值**（10 分）

#### **任务 1.3：假设误差项服从独立异方差**
- 计算 **系数的方差** 和 **T 值**（10 分）

#### **任务 1.4：假设误差项存在组内相关性（聚类到 city-level）**
- 计算 **系数的方差（组内异方差聚类）** 和 **T 值**（10 分）

#### **任务 1.5：使用 `statsmodels.api` 对上述三种情况进行检验**
- 采用 `import statsmodels.api as sm` 进行检验，并比较结果（10 分）

---

### **任务二：计算广义矩估计（GMM）的系数方差和 T 值（共 50 分）**

### **广义矩估计（GMM）**
**两步法估计公式**：$\hat{\beta}_{gmm} = (X' Z \Phi Z' X)^{-1} X' Z \Phi Z' y$
- 其中，**最优权重矩阵** 分两步计算：
  - 第一步：$\hat{\epsilon}^{(1)}$ 在 $\Phi = (Z'Z)^{-1}$ 的假设下估计得到
  - 第二步：$\Phi = (Z'\hat{\epsilon}^{(1)} \hat{\epsilon}^{(1)'}Z)^{-1}$

**估计结果**：
$\hat{\ln(wage)} = 0.0477 + 0.0614 \cdot educ + 0.0451 \cdot exper - 0.0009 \cdot expersq$

---

#### **任务 2.1：计算系数估计值**
- 估计 **$\beta$ 系数**，即使用 GMM 方法求解 **回归系数估计值**（10 分）

#### **任务 2.2：假设误差项服从独立同方差**
- 计算 **系数的方差** 和 **T 值**（10 分）

#### **任务 2.3：假设误差项服从独立异方差**
- 计算 **系数的方差** 和 **T 值**（10 分）

#### **任务 2.4：假设误差项存在组内相关性（聚类到 city-level）**
- 计算 **系数的方差（组内异方差聚类）** 和 **T 值**（10 分）

#### **任务 2.5：使用 `statsmodels.api` 对上述三种情况进行检验**
- 采用 `import statsmodels.api as sm` 进行检验，并比较结果（10 分）

---